subsys/usb/class/usb_dfu.c

```diff
@@ -298,10 +298,17 @@ struct dfu_data_t {
 	uint16_t bwPollTimeout;
 };
 
+#if FLASH_AREA_LABEL_EXISTS(image_1)
```

Comment:
This should also be applied to the descriptors

```diff
diff --git a/subsys/usb/class/usb_dfu.c b/subsys/usb/class/usb_dfu.c
index 2c22d4e414..e3a0d3ffb4 100644
--- a/subsys/usb/class/usb_dfu.c
+++ b/subsys/usb/class/usb_dfu.c
@@ -123,7 +123,9 @@ struct dev_dfu_mode_descriptor {
        struct usb_cfg_descriptor cfg_descr;
        struct usb_sec_dfu_config {
                struct usb_if_descriptor if0;
+#if FLASH_AREA_LABEL_EXISTS(image_1)
                struct usb_if_descriptor if1;
+#endif
                struct dfu_runtime_descriptor dfu_descr;
        } __packed sec_dfu_cfg;
 } __packed;
@@ -172,6 +174,7 @@ struct dev_dfu_mode_descriptor dfu_mode_desc = {
                        .bInterfaceProtocol = DFU_MODE_PROTOCOL,
                        .iInterface = 4,
                },
+#if FLASH_AREA_LABEL_EXISTS(image_1)
                .if1 = {
                        .bLength = sizeof(struct usb_if_descriptor),
                        .bDescriptorType = USB_INTERFACE_DESC,
@@ -183,6 +186,7 @@ struct dev_dfu_mode_descriptor dfu_mode_desc = {
                        .bInterfaceProtocol = DFU_MODE_PROTOCOL,
                        .iInterface = 5,
                },
+#endif
                .dfu_descr = {
                        .bLength = sizeof(struct dfu_runtime_descriptor),
                        .bDescriptorType = DFU_FUNC_DESC,
@@ -226,11 +230,13 @@ struct usb_string_desription {
                uint8_t bString[USB_BSTRING_LENGTH(FIRMWARE_IMAGE_0_LABEL)];
        } __packed utf16le_image0;
 
+#if FLASH_AREA_LABEL_EXISTS(image_1)
        struct image_1_descriptor {
                uint8_t bLength;
                uint8_t bDescriptorType;
                uint8_t bString[USB_BSTRING_LENGTH(FIRMWARE_IMAGE_1_LABEL)];
        } __packed utf16le_image1;
+#endif
 } __packed;
 
 USBD_STRING_DESCR_DEFINE(secondary)
@@ -267,6 +273,7 @@ struct usb_string_desription string_descr = {
                .bDescriptorType = USB_STRING_DESC,
                .bString = FIRMWARE_IMAGE_0_LABEL,
        },
+#if FLASH_AREA_LABEL_EXISTS(image_1)
        /* Image 1 String Descriptor */
        .utf16le_image1 = {
                .bLength = USB_STRING_DESCRIPTOR_LENGTH(
@@ -274,6 +281,7 @@ struct usb_string_desription string_descr = {
                .bDescriptorType = USB_STRING_DESC,
                .bString = FIRMWARE_IMAGE_1_LABEL,
        },
+#endif
 };
 
 /* This element marks the end of the entire descriptor. */
```

Reply:
@jfischer-phytec-iot Thanks for this diff, will apply it.

Reply:
Add this as well.
``` diff
diff --git a/subsys/usb/class/usb_dfu.c b/subsys/usb/class/usb_dfu.c
index 2c22d4e414..a9ca4b1354 100644
--- a/subsys/usb/class/usb_dfu.c
+++ b/subsys/usb/class/usb_dfu.c
@@ -679,10 +679,12 @@ static int dfu_custom_handle_req(struct usb_setup_packet *pSetup,
 				dfu_data.flash_area_id =
 				    FLASH_AREA_ID(image_0);
 				break;
+#if FLASH_AREA_LABEL_EXISTS(image_1)
 			case 1:
 				dfu_data.flash_area_id =
 				    UPLOAD_FLASH_AREA_ID;
 				break;
+#endif
 			default:
 				LOG_WRN("Invalid DFU alternate setting");
 				return -ENOTSUP;
```

Reply:
applied